3D Chess/Assets/Scripts/PieceEntity.cs

```diff
@@ -3,12 +3,15 @@
 
 public class PieceEntity : RealmObject
 {
+    // An `enum` cannot be saved directly in Realm, we need to use a type that can be saved which in this case is an `int`.
     public PieceType PieceType
     {
         get => (PieceType)Type;
         set => Type = (int)value;
```

Comment:
Should we have a setter for `Type`? I don't suppose we'll be changing the piece type ever.

Reply:
My suggestion would be no since `PieceEntity` should never change it's type. The only case where something like this could be helpful is when moving a pawn to the final row and turning into a different piece. I would handle this (very much like a real chess board) as a destroy on the pawn and an insert on a new piece (usually a queen) and therefore not offer changing the type at all.

Reply:
Sounds good, let's remove it then.

Reply:
a301a81 
It's private now (still need it when setting it during construction).